src/runtime/math.py

```diff
@@ -207,33 +207,114 @@ def radians(x: f64) -> f64:
     """
     return x * pi / 180.0
 
-def fabs(x: f64) -> f64:
+# fabs
+# supported data types: i32, i64, f32, f64
+@overload
+def fabs(x: f32) -> f32:
     """
     Return the absolute value of `x`.
     """
     if x < 0.0:
         return -x
     return x
 
+@overload
+def fabs(x: f64) -> f64:
+    """
+    Return the absolute value of `x`.
+    """
+    if x < 0.0:
+        return -x
+    return x
 
-def ldexp(x: f64, i: i32) -> f64:
-    return x * (2**i)
+@overload
+def fabs(x: i64) -> f64:
+    """
+    Return the absolute value of `x`.
+    """
+    if x < 0.0:
+        return -float(x)
+    return float(x)
 
+@overload
+def fabs(x: i32) -> f32:
+    """
+    Return the absolute value of `x`.
+    """
+    if x < 0.0:
+        return -float(x)
+    return float(x)
```

Comment:
Can `float(x)` return `f32` or `f64` depending on the kind of `x`? Or does it always return `f64`? @certik

Reply:
Well seems like it always returns `f64`,

https://github.com/lcompilers/lpython/blob/4163cd7d4fc3dcca1dc3eb8132ac6d4ce0761744/src/lpython/semantics/python_ast_to_asr.cpp#L3282-L3290

Reply:
Shall I report it as an issue? @czgdp1807

Reply:
@certik What do you think is the best way to do casting using `float`?

1. Retain the kind but change the type. That would mean `i32` will be converted to `f32`, `i64` to `f64` and like that. 
2. Always convert to `f64`. Then to convert `i32` to `f32` we will be compelled to use tricks like,

```python
result: f32 = float(x)
return result
```

Reply:
I think `float` has to always return f64, as default floating point numbers in CPython (in fact the only floating point numbers) are f64.

Then we already have automatic casting if you do:
```
a: f64
b: f32
a = 3.4444
b = a
```
Then `a` is f64, and in `b = a` it gets cast from f64 to f32.

So if you do: `result: f32 = float(x)`, then `float(x)` returns f64, and then it gets cast to f32.


In ASR `float` will also be a cast, `IntegerToReal 64`, then there will be another cast `RealToReal 64->32`. We can add an optimization pass that will "fuse" these two into just one "IntegerToReal 32". So that in Release mode there is no performance hit with the above design.

Reply:
I see. @Madhav2310 Go with the idea in point 2 of [my comment](https://github.com/lcompilers/lpython/pull/581#discussion_r903316579) and update this PR accordingly.